test/AutoRest.TestServer.Tests/body-complex.cs

```diff
@@ -23,16 +26,934 @@ public class BodyComplexTest: TestServerTestBase
+                { string.Empty, null }
+            };
+            Assert.AreEqual(content, result.Value.DefaultProgram);
+        });
+
+        [Test]
+        [Ignore("https://github.com/Azure/autorest.csharp/issues/289 https://github.com/Azure/autorest.csharp/issues/311")]
+        public Task PutComplexDictionaryValid() => TestStatus(async (host, pipeline) =>
+        {
+            var value = new DictionaryWrapper();
+            var content = new Dictionary<string, string?>
+            {
+                { "txt", "notepad" },
+                { "bmp", "mspaint" },
+                { "xls", "excel" },
+                { "exe", string.Empty },
+                { string.Empty, null }
+            };
+            foreach (var item in content)
+            {
+                value.DefaultProgram.Add(item);
+            }
+            return await DictionaryOperations.PutValidAsync(ClientDiagnostics, pipeline, value, host);
+        });
+
+        [Test]
+        public Task GetComplexDictionaryEmpty() => Test(async (host, pipeline) =>
+        {
+            var result = await DictionaryOperations.GetEmptyAsync(ClientDiagnostics, pipeline, host);
+            Assert.AreEqual(new Dictionary<string, string?>(), result.Value.DefaultProgram);
+        });
+
+        [Test]
+        [Ignore("https://github.com/Azure/autorest.csharp/issues/311")]
+        public Task PutComplexDictionaryEmpty() => TestStatus(async (host, pipeline) =>
+        {
+            var value = new DictionaryWrapper();
+            return await DictionaryOperations.PutValidAsync(ClientDiagnostics, pipeline, value, host);
+        });
+
+        [Test]
+        [Ignore("https://github.com/Azure/autorest.csharp/issues/289")]
+        public Task GetComplexDictionaryNull() => Test(async (host, pipeline) =>
+        {
+            var result = await DictionaryOperations.GetNullAsync(ClientDiagnostics, pipeline, host);
+            Assert.AreEqual(200, result.GetRawResponse().Status);
+            Assert.AreEqual(new Dictionary<string, string?>(), result.Value.DefaultProgram);
+        });
+
+        [Test]
+        public Task GetComplexDictionaryNotProvided() => Test(async (host, pipeline) =>
+        {
+            var result = await DictionaryOperations.GetNotProvidedAsync(ClientDiagnostics, pipeline, host);
+            Assert.AreEqual(200, result.GetRawResponse().Status);
+            Assert.AreEqual(new Dictionary<string, string?>(), result.Value.DefaultProgram);
+        });
+
+        [Test]
+        [Ignore("https://github.com/Azure/autorest.csharp/issues/312")]
+        public Task GetComplexInheritanceValid() => Test(async (host, pipeline) =>
+        {
+            var result = await InheritanceOperations.GetValidAsync(ClientDiagnostics, pipeline, host);
+            //"breed":"persian","color":"green","hates":[{"food":"tomato","id":1,"name":"Potato"},{"food":"french fries","id":-1,"name":"Tomato"}],"id":2,"name":"Siameeee"
```

Comment:
Commented code in the tests so that, once the issues are fixed, I don't have to refer to the javascript or swagger to have the object information.